### ISSUE TITLE
[feature] Remove unreachable instances from the database

### ISSUE THREAD

**Issue by mirabilos** [superseriousbusiness/gotosocial#2493]
### Is your feature request related to a problem ?

Sometimes, instances go away ([example](http://outdoors.lgbt/)); sometimes these are reconstructed, or new ones are constructed on the old name, but most of the time, they just go away.

In all those cases, however, my database retains entries (profiles, statūs, etc.) from them, and only in the “the instance is reconstructed” case they keep validity.

Similarly, if another instance defederates from me or I from them, my database’s entries about them will then be useless.

### Describe the solution you'd like.

Add an admin command to get rid of all information associated with a named instance.

Should that instance then connect with GtS again (and is in the allowlist or not in the blocklist), the database will slowly fill with info again as if it would be first seen, but otherwise, the space will be freed.

### Describe alternatives you've considered.

I’ve been told to please not manually muck with the database, so I’m not considering alternatives at the moment.

**Comment by tsmethurst** [comment]
You can do most of this already by suspending domains that have disappeared. Stub accounts will still be held in the database but all statuses, media, relationships, faves, etc will be deleted.

**Comment by mirabilos** [comment]
tobi dixit:

>You can do most of this already by suspending domains that have
>disappeared. Stub accounts will still be held in the database but all
>statuses, media, relationships, faves, etc will be deleted.

Ah. (“suspending” = setting a domain block?)

Might be useful to have a way, later, to clean up those stub
accounts as well, though.

Thanks,
//mirabilos

**Comment by tsmethurst** [comment]
> Ah. (“suspending” = setting a domain block?)

Yep!

> Might be useful to have a way, later, to clean up those stub accounts as well, though.

Yeah good idea, something like a "purge" command.

**Comment by mirabilos** [comment]
> > Ah. (“suspending” = setting a domain block?)
>
> Yep!

Actually, I could probably block them and then immediately unblock?

**Comment by tsmethurst** [comment]
> Actually, I could probably block them and then immediately unblock?

Yes, you could do that :) Then they can still federate again eventually if they come back up.
